src/app/features/habit/habitSlice.ts

```diff
@@ -0,0 +1,48 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { PayloadAction } from "@reduxjs/toolkit";
+import { Storage } from "../../../utilities/storage";
+import { HabitSlice } from "./habitTypes";
+
+const storage = new Storage();
+
+const initialHabit = {
+  id: null,
+  goal: "",
+  schema: "",
+  rewards: [],
+  habitStrength: {
+    strength: 0,
+    history: [],
+  },
+  data: [],
```

Comment:
What is the difference between `data` and `history`. Now the `data` contains all of the dates, so it is a history, isn't it?

Reply:
`habitStrenght` will be measured weekly/every few days with a separate questionnaire. So it will be different from habit data. Habit data give you only information about the habit in a single day.